lib/theme-api-client.js

```diff
@@ -156,11 +156,20 @@ function postTheme(options, callback) {
         }
 
         if (response.statusCode === 409) {
-            return callback(null, { themeLimitReached: true });
+            // This status code is overloaded, so we need to check the message to determine
+            // if we want to trigger the theme deletion flow.
+            const uploadLimitPattern = /You have reached your upload limit/;
+            if (payload && payload.title && uploadLimitPattern.exec(payload.title)) {
+                return callback(null, { themeLimitReached: true });
+            }
+
+            const message = 'You already have an theme upload in progress. You will need to wait for it to finish processing before uploading a new one.';
```

Comment:
🍹 a theme upload instead of an?  maybe it's just me but sounds wierd

Reply:
Good catch. This used to say "an upload".